### ISSUE TITLE
[L4D2] Mic doesn't work

### ISSUE THREAD

**Issue by pizzadude** [ValveSoftware/steam-for-linux#2390]
Mic works in Steam, TF2, etc, but not L4D2.
Testing mic in L4D2 doesn't work either.
Everything else works as expected.

```
Processor Information:
Vendor:  GenuineIntel
Speed: 3201 Mhz
4 logical processors
4 physical processors
HyperThreading:  Unsupported
FCMOV:  Supported
SSE2:  Supported
SSE3:  Supported
SSSE3:  Supported
SSE4a:  Unsupported
SSE41:  Supported
SSE42:  Supported

Network Information:
Network Speed:  

Operating System Version:
Linux Mint 13 Maya (32 bit)
Kernel Name:  Linux
Kernel Version:  3.2.0-41-generic-pae
X Server Vendor:  The X.Org Foundation
X Server Release:  11103000
X Window Manager:  Muffin
Steam Runtime Version:  steam-runtime-release-i386_2013-04-16

Video Card:
Driver:  NVIDIA Corporation GeForce GTX 660/PCIe/SSE2

Driver Version:  4.3.0 NVIDIA 310.14
Desktop Color Depth: 24 bits per pixel
Monitor Refresh Rate: 60 Hz
VendorID:  0x10de
DeviceID:  0x11c0
Number of Monitors:  1
Number of Logical Video Cards:  1
Primary Display Resolution:  1920 x 1080
Desktop Resolution: 1920 x 1080
Primary Display Size: 20.04" x 11.26"  (22.95" diag)
                                        50.9cm x 28.6cm  (58.3cm diag)
Primary Bus: PCI Express 16x
Primary VRAM: 2048 MB
Supported MSAA Modes:  2x 4x 8x 16x 

Sound card:
Audio device: VIA VT1708S

Memory:
RAM:  8015 Mb

Miscellaneous:
UI Language:  English
LANG:  en_CA.UTF-8
Microphone:  Not set
Total Hard Disk Space Available:  930900 Mb
Largest Free Hard Disk Block:  711187 Mb

Installed software:

Recent Failure Reports:
```

**Comment by grahammitchell** [comment]
Same problem. USB Logitech c920 webcam mic doesn't work in Left4Dead2 in-game (or in the lobby). It is detected and works in Steam generally. Doesn't matter whether I use push-to-talk or open mic, it never plays. Speech bubble appears over my head when I push-to-talk, but no sound. I can hear Windows teammates, and they can hear each other.

$ uname -a
Linux [redacted] 3.16.0-29-generic #39-Ubuntu SMP Mon Dec 15 22:27:29 UTC 2014 x86_64 x86_64 x86_64 GNU/Linux

**Comment by tnbp** [comment]
Everybody who has problems with a second mic (i.e. that is not connected to their main sound card, like a webcam mic or a USB headset) and is running pulseaudio should check if L4D2 is actually getting its input from that device - or if it is instead using the sound card.
Run pavucontrol (if you don't have it: it's probably in a package called pulseaudio-tools or something), go to the "Recording" tab (with L4D2 running in the background, of course!) and set it to capture from your desired input device.
If that fixes it, it is not a problem with L4D2.
Hope this helps.

**Comment by grahammitchell** [comment]
In my case, Steam is configured to capture audio from the USB webcam microphone, and that microphone tests correctly in Steam. If Left 4 Dead 2 ignores the Steam setting for audio input, that *is* a problem with L4D2.

**Comment by tnbp** [comment]
L4D2 and Steam are different clients to PulseAudio.

**Comment by grahammitchell** [comment]
It's still a L4D2 bug. It's a Steam game. L4D2 run in Steam should respect the audio source as configured in Steam settings.

But just for the record, the PulseAudio workaround worked for me. Thanks for the suggestion!

**Comment by g105b** [comment]
+1 this affects most steam games for me

**Comment by FinesseRus** [comment]
I have the same problem on OS X 10.11.2

**Comment by FinesseRus** [comment]
I have the same problem on OS X 10.11.2. I've found what the reason but I don't know how to solve it. Here what I've found:
* When you enter the game the system microphone input volume is set to zero (see screenshot).
* If you set it to normal level, quit the game and start it again the level will be set to zero again.

![2015-12-14 17 52 25](https://cloud.githubusercontent.com/assets/9006227/11775567/52eaf61e-a283-11e5-8b76-b7406465653d.png)
